test/dump.ts

```diff
@@ -0,0 +1,84 @@
+    it(`dumpText(${values})`,() => roundTripTest(ion.dumpText, values, expectedValues));
+    it(`dumpPrettyText(${values})`, () => roundTripTest(ion.dumpPrettyText, values, expectedValues));
+}
+
+// Dumps the provided values to Ion using the specified dump*() function, uses loadAll() to read the values
+// back in, then compares the input values to the loaded values.
+function roundTripTest(dumpFn: (...values: any[]) => any,
+                       inputValues: any[],
+                       expectedValues: dom.Value[]) {
+    let actualValues: dom.Value[] = dom.loadAll(dumpFn(inputValues));
+    let testComparisons: [dom.Value, dom.Value][] = actualValues.map(
+        (actualValue, index) => [actualValue, expectedValues[index]]
+    );
+    for (let [actualValue, expectedValue] of testComparisons) {
+        //TODO: This tests for a very weak common definition of equality.
+        //      We should make this a stronger test when .equals() is added to the Value interface[1].
+        //      In the meantime, the core writing/equality logic is more strictly tested by
+        //      test/dom/Value.ts, which evaluates the data produced by the Value.writeTo() function
+        //      at the core of the ion.dump*() methods.
+        //      [1] https://github.com/amzn/ion-js/issues/576
+        assert.deepEqual(actualValue.getAnnotations(), expectedValue.getAnnotations());
+        assert.equal(actualValue.isNull(), expectedValue.isNull());
+        assert.equal(actualValue.getType(), expectedValue.getType());
+    }
+}
+
+describe('dump*()', () => {
+    describe('Round tripping JS values', () => {
+        roundTripTests(exampleJsValuesWhere())
+    });
+    describe('Round tripping Ion values', () => {
+        roundTripTests(exampleIonValuesWhere())
+    });
+    it('Text stream', () => {
+        let ionText = ion.dumpText(null, 7, "Hello", [1, 2, 3]);
+        assert.equal(ionText, 'null\n7\n"Hello"\n[1,2,3]');
+    });
+    it('Pretty text stream', () => {
+        let ionPrettyText = ion.dumpPrettyText(null, 7, "Hello", [1, 2, 3]);
+        assert.equal(ionPrettyText, 'null\n7\n"Hello"\n[\n  1,\n  2,\n  3\n]');
+    });
+    it('Binary stream', () => {
+        let ionBinary: Uint8Array = ion.dumpBinary(null, 7, "Hello", [1, 2, 3]);
+        let writer = ion.makeBinaryWriter();
+        writer.writeNull(IonTypes.NULL);
+        writer.writeInt(7);
+        writer.writeString("Hello");
+        writer.stepIn(IonTypes.LIST);
+        writer.writeInt(1);
+        writer.writeInt(2);
+        writer.writeInt(3);
+        writer.stepOut();
+        writer.close();
+        let expectedBytes: Uint8Array = writer.getBytes();
+        let byteComparisons: [number, number][] = Array.from(ionBinary).map(
+            (byte, index) => [byte, expectedBytes[index]]
+        );
+        for (let [actualByte, expectedByte] of byteComparisons) {
+            assert.equal(actualByte, expectedByte);
+        }
```

Comment:
consider simplifying to `assert.deepEqual(ionBinary, expectedBytes)`, which i think accomplishes the same thing

Reply:
Based on serialization peculiarities I've experienced using `Uint8Array`, I had assumed that `deepEqual` would evaluate properties of the object itself rather than comparing the stored bytes. I'm pleased to discover that I was wrong. I've converted this to `deepEqual`, thanks.